gammapy/irf/irf_reduce.py

```diff
@@ -150,3 +155,44 @@ def make_mean_edisp(
     irf_stack.stack_edisp()
 
     return irf_stack.stacked_edisp
+
+
+def apply_containment_fraction(aeff, psf, radius):
+    """ Estimate PSF containment inside a given radius and correct effective area for leaking flux.
+
+    The PSF and effective area must have the same binning in energy.
+
+    Parameters
+    ----------
+    aeff : `~gammapy.irf.EffectiveAreaTable`
+        the input 1D effective area
+    psf : `~gammapy.irf.EnergyDependentPSFTable`
+        the input 1D PSF
+    radius : `~astropy.coordinates.Angle`
+        the maximum angle
+
+    Returns
+    -------
+    correct_aeff : `~gammapy.irf.EffectiveAreaTable`
+        the output corrected 1D effective area
+    """
+    center_energies = aeff.energy.nodes
+    containment = []
+    for index, energy in enumerate(center_energies):
+        try:
+            cont_ = psf.integral(energy, 0.0 * u.deg, radius)
+        except:
```

Comment:
I'd prefer to simplify this code section a bit:

- Remove the `enumerate` and just do the loop over energies
- Add the exception type that is thrown
- Remove the `finally` section and just add the `.append()` as the last statement in the loop

Reply:
Done except the exception type.
As far as I can see, the exception thrown has no type. It comes from the `UnivariateSpline` constructor when the `PSFTable._dp_domega` is bad (e.g. contains `nan`).
A solution might be to raise an exception in the constructor of `TablePSF` but we probably need an `except` without exception type somewhere, no?

Reply:
Yesterday I merged #2019. Once you've re-based, you can replace the loop over energy bins and just use `EnergyDependentTablePSF.containment(radius)`. This should also get rid of the `try` and `except` statement, as the new `EnergyDependentTablePSF.containment_radius()` should behave more stable.

Reply:
I just did the change to adapt to PR#2019.
For some reason the result of the integral has changed for the disk-like PSF I was using. There is a 5% effect on the test that was previously yielding the expected value. I need to understand why.

Reply:
Yesterday I discovered a minor bug in the new containment computation method, that I implemented: the rad vector was shifted by one pixel. I guess this could lead to the 5% effect you saw. I'm preparing a PR to fix this.

Reply:
I just merged #2022. Can you try and see if it fixes the 5% change you see?